core/src/main/java/org/openedx/core/domain/model/CourseDatesBannerInfo.kt

```diff
@@ -0,0 +1,68 @@
+package org.openedx.core.domain.model
+
+import org.openedx.core.R
+import org.openedx.core.domain.model.CourseBannerType.BLANK
+import org.openedx.core.domain.model.CourseBannerType.INFO_BANNER
+import org.openedx.core.domain.model.CourseBannerType.RESET_DATES
+import org.openedx.core.domain.model.CourseBannerType.UPGRADE_TO_GRADED
+import org.openedx.core.domain.model.CourseBannerType.UPGRADE_TO_RESET
+
+data class CourseDatesBannerInfo(
+    private val missedDeadlines: Boolean,
+    private val missedGatedContent: Boolean,
+    private val verifiedUpgradeLink: String,
+    private val contentTypeGatingEnabled: Boolean,
+    private val hasEnded: Boolean
+) {
+    val bannerType by lazy { getCourseBannerType() }
+
+    val headerResId = when (bannerType) {
+        RESET_DATES -> R.string.core_dashboard_banner_reset_date_header
+        else -> 0
+    }
+
+    val bodyResId = when (bannerType) {
+        INFO_BANNER -> R.string.core_dates_info_banner
+        UPGRADE_TO_GRADED -> R.string.core_dates_banner_upgrade_to_graded
+        UPGRADE_TO_RESET -> R.string.core_dates_banner_upgrade_to_reset
+        RESET_DATES -> R.string.core_dashboard_banner_reset_date_body
+        else -> 0
+    }
+
+    val buttonResId = when (bannerType) {
+        RESET_DATES -> R.string.core_dates_banner_reset_date_button
+        else -> 0
+    }
+
+    fun isBannerAvailableForUserType(isSelfPaced: Boolean): Boolean {
+        if (hasEnded) return false
+
+        val bannerType = getCourseBannerType()
+        val selfPacedAvailable = isSelfPaced && bannerType != BLANK
+        val instructorPacedAvailable = !isSelfPaced && bannerType == UPGRADE_TO_GRADED
+
+        return selfPacedAvailable || instructorPacedAvailable
+    }
+
+    private fun getCourseBannerType(): CourseBannerType = when {
+        isUpgradeToGradedBanner() -> UPGRADE_TO_GRADED
```

Comment:
please update the name as per implementation. The banner type will be used to take action from the user. i.e, 
```
canUpgradToGraded, canUpgradeToReset, canResetDates & infoBanner